### DISCUSSION TITLE
Switch back from moshi-ir to KSP with less code

### DISCUSSION THREAD

**Discussion by beigirad** [ZacSweers/MoshiX#685] [Q&A] [answered]
In moshi-ir readme, you said:

> If you have any issues, you can always fall back to Kapt/KSP

But didn't say how we can achieve this target with less code (in modular project with +40 module).

in root project `build.gradle` I added these lines but adapters doesn't generate. also I'm not sure that ` enabled = false` is working or not (cause there's no talk about this flag in documents)

```
allprojects {
    afterEvaluate {
        extensions.findByType<MoshiPluginExtension>()?.apply {
            enabled = false

            dependencies.add("implementation", "${libs.moshi.get()}")
            dependencies.add("ksp", "${libs.moshiCodeGenerator.get()}")
        }
}
```

**Comment by ZacSweers** [comment, marked as answer]
When I wrote that, I meant you can just not use moshi-ir and use KSP instead. This project doesn't seek to offer a gradle plugin to automatically switch between moshi's native KSP and moshi-ir.

**Reply by beigirad** [reply]
> doesn't seek to offer a gradle plugin

Thanks. Yeah, you're right. But providing an easy way to falling back will help me to use moshi-ir with peace of mind. (It was a suggestion)



In fact I wrote something that passes tests but doesn't work in runtime. To be sure that moshi-ir is the not problem I decided to falling back with minimum code change...

**Reply by ZacSweers** [reply]
Sorry the answer on the first part is no.

For the second - impossible to help given the limited information you've provided.
